sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_models.py

```diff
@@ -67,6 +70,12 @@ def __init__(self):
         self.metadata = None
         self.deleted = None
         self.deleted_version = None
+        default_encryption_scope = kwargs.get('x-ms-default-encryption-scope')
+        if default_encryption_scope:
+            self.encryption_scope = FileSystemEncryptionScope(
+                default_encryption_scope=default_encryption_scope,
+                prevent_encryption_scope_override=kwargs.get('x-ms-deny-encryption-scope-override', False)
+            )
```

Comment:
I'm a little weary on what is going on here? Was this stuff copied from `ContainerProperties`? The container property is called `default-encryption-scope` but the one on files and directories is just `encryption-scope`? I guess that could make sense, just want to be sure.

Reply:
Yup, this is 1:1 pattern-matched from `ContainerProperties` in `_models.py` for blob.

Reply:
I think this is probably needed to setup the default encryption scope in the properties but it's a bit odd that `encrytion_scope` here is a `FileSystemEncryptionScope` object but for directories and files it's a string but I guess it's because we can have that extra `x-ms-deny-encryption-scope-override`? 

I guess if this is how it is in blob, it's probably fine, but it's odd :). 

As a side note, did you do any testing with `prevent_encryption_scope_override`? I think we should at least try setting that when creating a file system and making sure this gets populated correctly? Doesn't have to be a test we check in, just something to try.

Reply:
Sounds good, will conduct some ad-hoc testing on the override!